### PR TITLE
changes made to filenames + references

### PR DESCRIPTION
**Pull request by Damon-Ma-MCHP**
_What are the reasons/motivation for this change?_
- Replaced any textual instances of "mchp" to "microchip" and "pf" to "polarfire" to keep consistency among Yosys naming schemes.

_Explain how this is achieved._
- Find and replace tool + manual replacement.

_If applicable, please suggest to reviewers how they can test the change._
- Build Yosys to make sure that there are no compile errors.

### CODE REVIEW

File: passes/pmgen/Makefile.inc [hunk 1]
```diff
@@ -37,14 +37,14 @@ $(eval $(call add_extra_objs,passes/pmgen/xilinx_dsp_cascade_pm.h))
...
 
-OBJS += passes/pmgen/mchp_dsp.o
-GENFILES += passes/pmgen/mchp_dsp_pm.h
-GENFILES += passes/pmgen/mchp_dsp_CREG_pm.h
-GENFILES += passes/pmgen/mchp_dsp_cascade_pm.h
-passes/pmgen/mchp_dsp.o: passes/pmgen/mchp_dsp_pm.h passes/pmgen/mchp_dsp_CREG_pm.h passes/pmgen/mchp_dsp_cascade_pm.h
-$(eval $(call add_extra_objs,passes/pmgen/mchp_dsp_pm.h))
-$(eval $(call add_extra_objs,passes/pmgen/mchp_dsp_CREG_pm.h))
-$(eval $(call add_extra_objs,passes/pmgen/mchp_dsp_cascade_pm.h))
+OBJS += passes/pmgen/microvhip.o
```
Comment (tony-min-1) [inline passes/pmgen/Makefile.inc]: typo?

Comment (tony-min-1) [reply passes/pmgen/Makefile.inc]: should be `microchip_dsp.o` ?

Comment (Damon-Ma-MCHP) [reply passes/pmgen/Makefile.inc]: yeah, fixed it right now, Yosys is still building so I'm just watching out for other mistakes I may have made.

File: techlibs/microchip/synth_microchip.cc
Comment (tony-min-1) [inline techlibs/microchip/synth_microchip.cc:1]: line 115 
`family = "pf"` should be updated as well

**Pull request merged**